test/test_torch.py

```diff
@@ -17596,6 +17596,29 @@ def test_multinomial_empty(self, device):
             out = torch.multinomial(probs, num_samples=num_samples, replacement=replacement)
             self.assertEqual(out, expected)
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(*(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes(include_bfloat16=False)))
+    def test_count_nonzero(self, device, dtype):
+        for ndims in (0, 2, 1, 3):
+            shape = self._rand_shape(ndims, 5, 10)
+            for n in range(ndims + 1):
+                for c in combinations(list(range(ndims)), n):
+                    for count_dim in permutations(c):
+                        if shape == ():
+                            x = torch.tensor(())
```

Comment:
When constructing the tensor you still need to pass device=device and dtype=dtype

Reply:
Now `x` is passed as `.tolist()` which will force it to take the else which considers `dtype` and `device`.

https://github.com/pytorch/pytorch/blob/18fe9d267c269851ba616f4e47120e5306c9827b/torch/testing/_internal/common_utils.py#L867-L873

Reply:
Also there's one thing I failed to ask earlier: will NaN and inf work as expected? Is it worth adding a small separate test that just tests values (NaN, +inf, -inf) and makes sure we handle these extremals like NumPy does?

Reply:
Right makes sense.